### PR TITLE
GEODE-4858: Refactor destroy gateway-receiver command

### PR DESCRIPTION
**Pull request by pdxrunner**
Refactored to use SingleGfshCommand and ResultModel

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [X] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [X] Is your initial contribution a single, squashed commit?

- [X] Does `gradlew build` run cleanly?

- [X] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyGatewayReceiverCommand.java [hunk 1]
```diff
@@ -61,32 +59,19 @@ public Result destroyGatewayReceiver(
     Set<DistributedMember> members = getMembers(onGroups, onMember);
 
     List<CliFunctionResult> functionResults =
-        executeAndGetFunctionResult(DestroyGatewayReceiverFunction.INSTANCE, ifExists, members);
+        executeAndGetFunctionResult(DestroyGatewayReceiverFunction.INSTANCE, null, members);
 
-    CommandResult result = ResultBuilder.buildResult(functionResults);
+    ResultModel result = ResultModel.createMemberStatusResult(functionResults, ifExists);
+    result.setConfigObject(new CacheConfig.GatewayReceiver());
```
Comment (sboorlagadda) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyGatewayReceiverCommand.java]: May be you could pass `null` as it is not required to be used in updateClusterConfig?

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/functions/DestroyGatewayReceiverFunction.java [hunk 2]
```diff
@@ -25,6 +26,7 @@
 import org.apache.geode.internal.cache.execute.InternalFunction;
 import org.apache.geode.internal.logging.LogService;
 import org.apache.geode.management.internal.cli.CliUtil;
+import org.apache.geode.management.internal.cli.functions.CliFunctionResult.StatusState;
 
 public class DestroyGatewayReceiverFunction implements InternalFunction {
```
Comment (sboorlagadda) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/functions/DestroyGatewayReceiverFunction.java]: To be aligned with other refactoring, I would request to extend CliFunction

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyGatewayReceiverCommand.java (hunk 1 above)
Comment (pdxrunner) [reply geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyGatewayReceiverCommand.java]: Thanks. I missed this change on my first refactoring iteration.

**Comment by pdxrunner** [comment]
Precheckin is green

**Pull request merged**